### PR TITLE
Another round of VM microoptimizations

### PR DESCRIPTION
**Pull request by roman-khimov**
Some nitpicking.

### CODE REVIEW

File: pkg/vm/stack.go [hunk 1]
```diff
@@ -170,6 +170,11 @@ func (s *Stack) Push(e Element) {
 	s.refs.Add(e.value)
 }
 
+// PushItem pushed an Item to the stack.
+func (s *Stack) PushItem(i stackitem.Item) {
```
Comment (AnnaShaleva) [inline pkg/vm/stack.go:174]: Don't we also want to replace all `PushVal` usages inside the `pkg/core/interop_system.go`?

File: pkg/vm/vm.go [hunk 2]
```diff
@@ -1562,7 +1570,7 @@ func (v *VM) handleException() {
 				continue
 			}
 			for i := 0; i < pop; i++ {
-				ctx := v.istack.Pop().Value().(*Context)
+				ctx := v.istack.Pop().value.(*Context)
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/vm/vm.go:1578]: Fixed.

File: pkg/vm/stack.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/vm/stack.go:174]: OK, let's try that too.

**Comment by roman-khimov** [comment]
It's a theory, btw, but I think the main problem with `PushVal` is not the type switch itself, but rather that it prevents aggressive code inlining that can easily happen now with trivial type conversions/short initializations.

**Review by AnnaShaleva — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]

**Pull request merged**